sdk/appconfiguration/azure-data-appconfiguration/README.md

```diff
@@ -135,14 +137,60 @@ ConfigurationAsyncClient client = new ConfigurationClientBuilder()
         .buildAsyncClient();
 ```
 
-You can also use `TokenCredential` to create a configuration client, such as an Azure Active Directory (AAD) token.
-Unlike a connection string if you're using an AAD token you must supply the endpoint of AppConfiguration service. The
-endpoint can be obtained by going to your App Configuration instance in the Azure portal and navigating to "Overview"
-page and look for the "Endpoint" keyword. 
+#### Use AAD token
+
+Here we demonstrate using [DefaultAzureCredential][default_cred_ref]
+to authenticate as a service principal. However, the configuration client
+accepts any [azure-identity][azure_identity] credential. See the
+[azure-identity][azure_identity] documentation for more information about other
+credentials.
+
+##### Create a service principal (optional)
+This [Azure CLI][azure_cli] snippet shows how to create a
+new service principal. Before using it, replace "your-application-name" with
+the appropriate name for your service principal.
+
+Create a service principal:
+```Bash
+az ad sp create-for-rbac --name http://my-application --skip-assignment
+```
+
+> Output:
+> ```json
+> {
+>     "appId": "generated app id",
+>     "displayName": "my-application",
+>     "name": "http://my-application",
+>     "password": "random password",
+>     "tenant": "tenant id"
+> }
+> ```
+
+Use the output to set **AZURE_CLIENT_ID** ("appId" above), **AZURE_CLIENT_SECRET**
+("password" above) and **AZURE_TENANT_ID** ("tenant" above) environment variables.
+The following example shows a way to do this in Bash:
+```Bash
+export AZURE_CLIENT_ID="generated app id"
+export AZURE_CLIENT_SECRET="random password"
+export AZURE_TENANT_ID="tenant id"
+```
+
+Assign one of the applicable [App Configuration roles][app_config_role] to the service principal.
+
+##### Create a client
+Once the **AZURE_CLIENT_ID**, **AZURE_CLIENT_SECRET** and
+**AZURE_TENANT_ID** environment variables are set,
+[DefaultAzureCredential][default_cred_ref] will be able to authenticate the
+configuration client.
+
+Constructing the client also requires your configuration store's URL, which you can
+get from the Azure CLI or the Azure Portal. In the Azure Portal, the URL can be found listed as the service "Endpoint"
 
 ```Java
-// An example of using TokenCredential and Endpoint to create a synchronous client
-TokenCredential credential = new DefaultAzureCredential();
+import com.azure.identity.DefaultAzureCredential;
```

Comment:
Prefer to use code embedding for READMEs.

Reply:
updated.